viewing_party/party.py

```diff
@@ -1,23 +1,183 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    '''
+    takes in a movie title, genre and rating
+    returns dictionary with the keys "title", "genre", and "rating"
+    with their respective values.
+    '''
+    movie = {}
+    if None in [title, genre, rating]: 
+        return None
+    movie["title"] = title
+    movie["genre"] = genre
+    movie["rating"] = rating
+    return movie
+
+def add_to_watched(user_data, movie):
+    '''
+    takes in user_data and movie
+    returns updated user_data with movie added to watched
+    '''
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    '''
+    takes in user_data and movie,
+    returns updated user_data with movie added to watchlist
+    '''
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    '''
+    takes user_data and title of a movie, 
+    if movie in watchlist, will add movie to watched and remove from watchlist
+    returns updated user_data
+    '''
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)

```

Comment:
This works in this particular case because we only expect the user to have one copy of each movie in their watchlist. But in general, try to avoid modifying a data structure as we're doing a for loop over it; very strange and hard to debug behavior can result. Consider instead using the loop to identify what needs to be removed, then remove it after the loop.